src/community/script/core/src/main/java/org/geoserver/script/rest/converter/ScriptConverter.java

```diff
@@ -0,0 +1,172 @@
+/** (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ *
+ * @author David Vick, Boundless 2017
+ **/
+package org.geoserver.script.rest.converter;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import com.thoughtworks.xstream.io.json.JettisonMappedXmlDriver;
+import org.geoserver.config.util.SecureXStream;
+import org.geoserver.rest.RequestInfo;
+import org.geoserver.rest.converters.XStreamCatalogListConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+/**
+ * This class enables us to extend the XMLXStreamListConverter and JSONXStreamListConverter classes so we can
+ * override the default XML and JSON output of the REST module.  Because these message converters have a lower
+ * priority than the converters in the REST module, they will always be called and therefore to prevent affecting
+ * the expected behavior of the REST module we are checking the request path and only customizing the output
+ * if the request is within the Scripts module.
+ */
+public abstract class ScriptConverter extends XStreamCatalogListConverter {
+
+    /**
+     * XML handling for Script lists
+     */
+    @Component
+    public static class XMLXStreamScriptListConverter extends XMLXStreamListConverter {
+        @Autowired
+        HttpServletRequest request;
+
+        @Override
+        protected XStream createXStreamInstance() {
+            return new SecureXStream();
+        }
+
+        @Override
+        public void encodeLink(String link, HierarchicalStreamWriter writer) {
+            String path = request.getRequestURI();
+            if (path.contains("script")) {
```

Comment:
Need to do a stricter check then that. What if there is a layer or store named "script" for some reason? Then doing a GET for that layer may use this converter.

Reply:
Created the following method
```
protected static boolean checkPath(HttpServletRequest request) {
        String path = request.getRequestURI();
        if (path.contains("script/apps") || (path.contains("script/function"))
                || (path.contains("script/wfs/tx")) || (path.contains("script/wps"))) {
            return true;
        } else {
            return false;
        }
    }
```